src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -0,0 +1,1218 @@
+     */
+    @VisibleForTesting
+    static List<? extends Locatable> getSortedExonAndStartStopPositions(final GencodeGtfTranscriptFeature transcript) {
+
+        // Sort by exon number first:
+        transcript.getExons().sort((lhs, rhs) -> lhs.getExonNumber() < rhs.getExonNumber() ? -1 : (lhs.getExonNumber() > rhs.getExonNumber() ) ? 1 : 0 );
+
+        final List<Locatable> exonList = new ArrayList<>(transcript.getExons().size());
+        for ( final GencodeGtfExonFeature exon : transcript.getExons() ) {
+
+            // Add in a CDS region:
+            if ( exon.getCds() != null ) {
+
+                // If we have a start codon that is not in the CDS for some reason,
+                // we need to add it to our list:
+                if (exon.getStartCodon() != null) {
+                    if ( !exon.getCds().contains(exon.getStartCodon()) ) {
+                        exonList.add( exon.getStartCodon() );
+                    }
+                }
+
+                exonList.add( exon.getCds() );
+
+                // If we have a stop codon that is not in the CDS for some reason,
+                // we need to add it to our list:
+                if (exon.getStopCodon() != null) {
+                    if ( !exon.getCds().contains(exon.getStopCodon()) ) {
+                        exonList.add( exon.getStopCodon() );
+                    }
+                }
+
+            }
+            else if (exon.getStartCodon() != null) {
+                exonList.add( exon.getStartCodon() );
+            }
+            else if ( exon.getStopCodon() != null ) {
+                exonList.add( exon.getStopCodon() );
+            }
+        }
+        return exonList;
+    }
+
+    /**
+     * Gets the {@link org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotation.VariantClassification} of the given {@code altAllele} for the given {@code variant}.
+     * @param variant The {@link VariantContext} to classify.
+     * @param altAllele The {@link Allele} of the given {@code variant} to classify.
+     * @param variantType The {@link org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotation.VariantType} of the given {@code variant}.
+     * @param exon The {@link GencodeGtfExonFeature} in which the given {@code variant} occurs.
+     * @param sequenceComparison The {@link org.broadinstitute.hellbender.tools.funcotator.FuncotatorUtils.SequenceComparison} for the given {@code variant}.
+     * @return A {@link org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotation.VariantClassification} based on the given {@code allele}, {@code variant}, {@code exon}, and {@code sequenceComparison}.
+     */
+    @VisibleForTesting
+    static GencodeFuncotation.VariantClassification getVariantClassification(final VariantContext variant,
```

Comment:
Wouldn't `createVariantClassification` be a more appropriate name?

Reply:
Yeah - good point.